tests/EntryPoints/REST/ApprovePageApiTest.php

```diff
@@ -78,33 +95,47 @@ public function testApprovalFailsForMissingPageId(): void {
 	}
 
 	public function testPageIsApproved(): void {
-		$approvalLog = new InMemoryApprovalLog();
 		$pageId = $this->getIdOfExistingPage( 'Page to be approved' );
 
 		$this->executeHandler(
-			$this->newApprovePageApi( $approvalLog ),
+			$this->newApprovePageApi(),
 			$this->createValidRequestData( $pageId )
 		);
 
 		$this->assertTrue(
-			$approvalLog->getApprovalState( $pageId )->isApproved
+			$this->approvalLog->getApprovalState( $pageId )->isApproved
 		);
 	}
 
 	public function testAPIUserIsInApprovalState(): void {
-		$approvalLog = new InMemoryApprovalLog();
 		$pageId = $this->getIdOfExistingPage( 'API User' );
 		$user = $this->mockRegisteredUltimateAuthority();
 
 		$this->executeHandler(
-			$this->newApprovePageApi( $approvalLog ),
+			$this->newApprovePageApi(),
 			$this->createValidRequestData( $pageId ),
 			authority: $user
 		);
 
 		$this->assertSame(
 			$user->getUser()->getId(),
-			$approvalLog->getApprovalState( $pageId )->approverId
+			$this->approvalLog->getApprovalState( $pageId )->approverId
+		);
+	}
+
+	public function testApprovedPageHtmlIsSaved(): void {
+		$pageId = $this->getIdOfExistingPage( 'Page to be saved' );
+
+		$this->executeHandler(
+			$this->newApprovePageApi(),
+			$this->createValidRequestData( $pageId )
+		);
+
+		$this->assertSame( <<<EOT
+<p>Whatever wikitext
```

Comment:
It is not so obvious where this wikitext is coming from, especially when reading this on GitHub

Reply:
Made explicit in https://github.com/ProfessionalWiki/PageApprovals/pull/32/commits/948e3b8f41439399f63e4792c1d1bb605335752a